sys/include/net/lora.h

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (C) 2017 Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    net_lora LoRa modulation
+ * @ingroup     net
+ * @brief       LoRa modulation header definitions
+ * @{
+ *
+ * @file
+ * @brief       LoRa modulation header definitions
+ *
+ * @author      Alexandre Abadie <alexandre.abadie@inria.fr>
+ */
+
+#ifndef NET_LORA_H
+#define NET_LORA_H
+
+#include <stdint.h>
+#include <stdlib.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @name    LoRa modulation default values
+ * @{
+ */
+#define LORA_FREQUENCY_RESOLUTION      (61.03515625)     /**< Frequency resolution in Hz */
+#define LORA_PREAMBLE_LENGTH           (8U)              /**< Preamble length, same for Tx and Rx */
+#define LORA_SYMBOL_TIMEOUT            (10U)             /**< Symbols timeout (s) */
+#define LORA_BW_DEFAULT                (LORA_BW_125_KHZ) /**< Set default bandwidth to 125kHz */
+#define LORA_SF_DEFAULT                (LORA_SF12)       /**< Set default spreading factor to 12 */
+#define LORA_CR_DEFAULT                (LORA_CR_4_8)     /**< Set default coding rate to 8 */
+#define LORA_FIX_LENGTH_PAYLOAD_ON     (false)           /**< Set fixed payload length on */
+#define LORA_IQ_INVERSION              (false)           /**< Set inverted IQ on */
+#define LORA_FREQUENCY_HOPPING         (false)           /**< Frequency hopping on */
+#define LORA_FREQUENCY_HOPPING_PERIOD  (0U)              /**< Frequency hopping period */
+#define LORA_FIXED_HEADER_LEN_MODE     (false)           /**< Set fixed header length mode (implicit header) */
+#define LORA_PAYLOAD_CRC_ON            (true)            /**< Enable payload CRC, optional */
+#define LORA_PAYLOAD_LENGTH            (0U)              /**< Set payload length, unused with implicit header */
+#define LORA_SYNCWORD_PUBLIC           (0x34)            /**< Syncword used for public networks */
+#define LORA_SYNCWORD_PRIVATE          (0x12)            /**< Syncword used for private networks */
+/** @} */
+
+/**
+ * @name    LoRa modulation available values
+ *
+ */
+/**
+ * @brief   LoRa modulation bandwidth.
+ */
+enum {
+    LORA_BW_125_KHZ = 0,               /**< 125 kHz bandwidth */
+    LORA_BW_250_KHZ,                   /**< 250 kHz bandwidth */
+    LORA_BW_500_KHZ                    /**< 500 kHz bandwidth */
```

Comment:
Are these values standardized? If yes, you should explicitly set them.

Reply:
No they aren't, same for other values. In the case of the SX127x radios, they are directly linked to #defined values (SX1276_RF_LORA_MODEMCONFIG1_BW_125_KHZ for examples).

Reply:
> In the case of the SX127x radios, they are directly linked to #defined values (SX1276_RF_LORA_MODEMCONFIG1_BW_125_KHZ for examples).

How?

Reply:
Answered the question myself (see https://github.com/RIOT-OS/RIOT/pull/7697#discussion_r144526946)